pkg/bytecode/value.go

```diff
@@ -189,3 +218,46 @@ func (m mapVal) Index(idx value) (value, error) {
 	}
 	return val, nil
 }
+
+type noneVal struct{}
+
+func (noneVal) Type() *parser.Type {
+	return parser.NONE_TYPE
+}
+
+func (noneVal) String() string {
+	return ""
+}
+
+func (noneVal) Equals(_ value) bool {
+	return false
+}
+
+func normalizeSliceIndices(start, end value, length int) (int, int, error) {
+	startIdx := 0
+	if _, ok := start.(numVal); ok {
+		startIdx = int(start.(numVal))
+	}
+	endIdx := length
+	if _, ok := end.(numVal); ok {
+		endIdx = int(end.(numVal))
+	}
```

Comment:
nit: I'd move this block of code to after the stuff with startIdx - keep all the startIdx and the endIdx stuff together.